sike.go

```diff
@@ -0,0 +1,153 @@
+// This file contains APIs for Supersingular Isogeny Key Encapsulation (SIKE) protocol
+// which is recently submitted to NIST PQC standardization workshop.
+// The underlying core functions are based on SIDH API developed by Cloudflare.
+// SIKE requires NIST's approved hash functions such as sha3 and cSHAKE256 to encapsulate
+// the key based on NIST specifications. 
+//
+//Author: Amir Jalali  				ajalali2016@fau.edu
+//Date: Feb 2018
+
+package p751sidh
+
+import (
+	"io"
+	"golang.org/x/crypto/sha3"
+	"bytes"
+)
+
+const (
+	// The message size, in bytes.
+	MessageSize = 32
```

Comment:
This code is specialized to p751 (as is the rest of the code in the p751sidh package). Since SIKE is a general construction that takes a SIDH implementation and builds upon it. Maybe a refactor of this repo so that sike takes a general SIDH struct would make sense. For example, https://golang.org/pkg/crypto/elliptic/#Curve has a Curve object and that is used in https://golang.org/pkg/crypto/ecdsa/#GenerateKey. This could make the package more generally useful once there are other SIDH implementations in Go other than p751sidh.